ui-trackoccupancydiagram/src/components/helpers/drawElements/drawOccupancyZonesTexts.ts

```diff
@@ -35,71 +41,111 @@ export const drawOccupancyZonesTexts = ({
   };
   arrivalTime: number;
   departureTime: number;
+  yPosition: number;
+  isThroughTrain: boolean;
+  selectedTrainId: string;
 }) => {
   const zoneOccupancyLength = departureTime - arrivalTime - STROKE_WIDTH;
 
   const isBelowBreakpoint = (breakpoint: keyof typeof BREAKPOINTS) =>
     zoneOccupancyLength < BREAKPOINTS[breakpoint];
 
-  const textLength = ctx.measureText(zone.originStation!).width;
-  const { xName, yName } = {
-    xName: isBelowBreakpoint('medium')
-      ? arrivalTime - textLength + STROKE_WIDTH
+  const originTextLength = ctx.measureText(zone.originStation!).width;
+  const nameTextLength = ctx.measureText(zone.arrivalTrainName).width;
+
+  const { xOriginName, yOriginName } = {
+    xOriginName: isBelowBreakpoint('medium')
+      ? arrivalTime - originTextLength + STROKE_WIDTH
       : arrivalTime + X_INITIAL_POSITION_OFFSET,
-    yName: isBelowBreakpoint('medium')
-      ? OCCUPANCY_ZONE_START - Y_MEDIUM_POSITION_OFFSET
-      : OCCUPANCY_ZONE_START - Y_INITIAL_POSITION_OFFSET,
+    yOriginName: isBelowBreakpoint('medium')
+      ? yPosition - Y_MEDIUM_POSITION_OFFSET
+      : yPosition - Y_INITIAL_POSITION_OFFSET,
+  };
+
+  const { xTrainName, yTrainName } = {
+    xTrainName: isBelowBreakpoint('medium')
+      ? arrivalTime - X_MEDIUM_POSITION_OFFSET_BACKGROUND
+      : arrivalTime,
+    yTrainName: isBelowBreakpoint('medium')
+      ? yPosition - Y_MEDIUM_POSITION_OFFSET_BACKGROUND
+      : yPosition - Y_INITIAL_POSITION_OFFSET_BACKGROUND,
```

Comment:
This structure might be more elegant, what do you think ?
```suggestion
const { xOriginName, yOriginName, xTrainName, yTrainName } = isBelowBreakpoint('medium')
    ? {
        xOriginName: arrivalTime - originTextLength + STROKE_WIDTH,
        yOriginName: yPosition - Y_MEDIUM_POSITION_OFFSET,
        xTrainName: arrivalTime - X_MEDIUM_POSITION_OFFSET_BACKGROUND,
        yTrainName: yPosition - Y_MEDIUM_POSITION_OFFSET_BACKGROUND,
      }
    : {
        xOriginName: arrivalTime + X_INITIAL_POSITION_OFFSET,
        yOriginName: yPosition - Y_INITIAL_POSITION_OFFSET,
        xTrainName: arrivalTime,
        yTrainName: yPosition - Y_INITIAL_POSITION_OFFSET_BACKGROUND,
      };
```

Reply:
yes much more, done